### DISCUSSION TITLE
How to set the homepage as the preview page or remove the homepage?

### DISCUSSION THREAD

**Discussion by uioonm** [web-infra-dev/rspress#1417] [Q&A] [answered]
我希望将/guide/start/index作为首页，或者将首页展示成预览页的形式，应该如何设置？
是否可以将rspress.dev的源码分享方便对其中的配置进行参考

**Comment by Timeless0911** [comment]
- The homepage is `index.md(x)`, you can import some component and write it as what you need.
- source code: https://github.com/web-infra-dev/rspress/tree/main/packages/document

**Reply by uioonm** [reply]
非常感谢你的回答，但是我希望首页是[https://rspress.dev/guide/start/introduction](url)的效果（可以直接看到所有文档并拥有两侧侧边栏），但是因为点击title或者icon会返回返回到首页，所以我必须设置一个“首页”，而首页貌似无法设置左侧侧边栏，我应该如何配置解决这个问题呢？或者能否实现下面两种解决方案的其中一种？
- 将首页设置成rspress提供的总览页[https://rspress.dev/guide/default-theme/overview-page](url)
- 设置title点击跳转的路径为指定路径

**Reply by Timeless0911** [reply]
我用中文说的清楚一点哈，首页取决于 `index.md(x)` 文件如何编写，不取决于配置，所写即所得。可以参考这个用例：https://github.com/web-infra-dev/rspress/blob/main/e2e/fixtures/auto-nav-sidebar

**Reply by uioonm** [reply]
很抱歉可能我表达的不是很清楚，我希望的是最好通过现有的功能直接实现而不是自己编写，rspress是否直接快速实现这三个功能呢？

- 首页是[https://rspress.dev/guide/start/introduction](https://github.com/web-infra-dev/rspress/discussions/url)的效果
- 将首页设置成rspress提供的总览页[https://rspress.dev/guide/default-theme/overview-page](https://github.com/web-infra-dev/rspress/discussions/url)
- 设置title和icon点击跳转的路径为指定路径

**Reply by Timeless0911** [reply]
我感觉你还是没理解，快速实现也是自己编写，可以自己跑跑 demo 哈

**Reply by uioonm** [reply]
你好我运行了[https://github.com/web-infra-dev/rspress/blob/main/e2e/fixtures/auto-nav-sidebar](url)的demo并未找到相应的解决方法，我尝试将api作为根路径但是这样就无法编译通过，貌似根路径的_meta.json只能用于配置nav栏无法设置overview: true或侧边栏，并且也未找到如何取消首页或者更改首页路径的方法，是否可以告知如何解决，非常感谢orz
（对于更改首页路径，我目前是通过直接重定向解决了但是这样会闪烁一下，想知道是否rspress本身就提供这个路径的设置）

**Reply by Timeless0911** [reply, marked as answer]
没有将某个路径作为根路径的操作，也没有取消首页的操作，首页希望是什么就在首页 index.md 写什么，无法配置，也无法凭空对非文件夹内容使用 overview。

你可以直接操作编译产物 html 文件进行重命名或修改，或者自己做客户端或者服务端的重定向，服务端重定向不会有闪烁，客户端重定向是否闪烁取决于你重定向的时机以及如何控制渲染的副作用。
